config/config_test.go

```diff
@@ -5,171 +5,211 @@ import (
 	"regexp"
 	"testing"
 
+	"github.com/stretchr/testify/suite"
+
 	"github.com/coreruleset/go-ftw/test"
 	"github.com/coreruleset/go-ftw/utils"
-	"github.com/stretchr/testify/assert"
 )
 
-var yamlConfig = `---
+var testData = map[string]string{
+	"TestNewConfigFromFile": `---
 logfile: 'tests/logs/modsec2-apache/apache2/error.log'
 testoverride:
   input:
     dest_addr: 'httpbingo.org'
     port: '1234'
   ignore:
     '920400-1$': 'This test must be ignored'
-`
-
-var yamlCloudConfig = `---
+`,
+	"TestNewConfigFromFileRunMode": `---
 mode: 'cloud'
-`
-
-var yamlBadConfig = `
+`,
+	"bad": `
 ---
 logfile: 'tests/logs/modsec2-apache/apache2/error.log'
 doesNotExist: ""
-`
+`,
+	"jsonConfig": `
+{"test": "type"}
+`,
+}
 
-var jsonConfig = `{"test": "type"}`
+type fileTestSuite struct {
+	suite.Suite
+	filename string
+	cfg      *FTWConfiguration
+}
 
-func TestNewDefaultConfig(t *testing.T) {
-	cfg := NewDefaultConfig()
-	assert.Equal(t, DefaultLogMarkerHeaderName, cfg.LogMarkerHeaderName)
-	assert.Equal(t, DefaultRunMode, cfg.RunMode)
-	assert.Equal(t, "", cfg.LogFile)
+type envTestSuite struct {
+	suite.Suite
 }
 
-func TestNewConfigBadFileConfig(t *testing.T) {
-	filename, _ := utils.CreateTempFileWithContent(jsonConfig, "test-*.yaml")
-	defer os.Remove(filename)
-	cfg, err := NewConfigFromFile(filename)
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
+type baseTestSuite struct {
+	suite.Suite
 }
 
-func TestNewConfigConfig(t *testing.T) {
-	filename, _ := utils.CreateTempFileWithContent(yamlConfig, "test-*.yaml")
+func TestConfigTestSuite(t *testing.T) {
+	suite.Run(t, new(baseTestSuite))
+	suite.Run(t, new(fileTestSuite))
+	suite.Run(t, new(envTestSuite))
+}
 
-	cfg, err := NewConfigFromFile(filename)
+func (s *fileTestSuite) SetupTest() {
+}
 
-	assert.NoError(t, err)
-	assert.NotNil(t, cfg)
-	assert.NotEmpty(t, cfg.TestOverride.Overrides, "Ignore list must not be empty")
+func (s *envTestSuite) SetupTest() {
+}
 
-	for id, text := range cfg.TestOverride.Ignore {
-		assert.Contains(t, (*regexp.Regexp)(id).String(), "920400-1$", "Looks like we could not find item to ignore")
-		assert.Equal(t, "This test must be ignored", text, "Text doesn't match")
+func (s *fileTestSuite) BeforeTest(_, name string) {
+	var err error
+	s.filename, _ = utils.CreateTempFileWithContent(testData[name], "test-*.yaml")
+	s.cfg, err = NewConfigFromFile(s.filename)
+	s.NoError(err)
+	s.NotNil(s.cfg)
+}
+
+func (s *fileTestSuite) AfterTest(_ string, _ string) {
```

Comment:
AFAICT, AfterTest and TearDownTest do the same thing, except that AfterTest has arguments. I suggest to use TearDownTest because we already use that and don't use the arguments.